src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/custom.py

```diff
@@ -51,3 +51,13 @@ def list_vm(client, resource_group, lab_name, order_by=None, top=None,
 
     return client.list(resource_group, lab_name,
                        expand=expand, filter=filters, top=top, order_by=order_by)
+
+
+def create_or_update_secret(client, resource_group, lab_name, name, value, user_name=None):
+    from .sdk.devtestlabs.models.secret import Secret
+
+    return client.create_or_update_resource(resource_group_name=resource_group,
+                                            lab_name=lab_name,
+                                            user_name=user_name,
+                                            name=name,
+                                            secret=Secret(value=value))
```

Comment:
You are right, we can definitely reflect and get the create / update of secrets. If we do that then based on the current model for the `Secret` class, I have to ignore bunch of parameters that doesn't make sense to user like `provisioning_state`, `unique_identifier`, `id`, `type`, `location`, `tags` because they are not required from the portal experience. I'd agree with you that that's the swagger modeling issue. 

Let me update the command based on direct reflection and ignore unnecessary param :)